Assets/Scripts/Game/UserInterface/ListBox.cs

```diff
@@ -542,6 +542,10 @@ public void AddItem(string text, out ListItem itemOut, int position = -1, object
                 font = DaggerfallUI.DefaultFont;
 
             TextLabel textLabel = new TextLabel();
+
+            textLabel.Parent = this; //Establish parent early.
+            _ = textLabel.Rectangle; //Call GetRectangle() early to establish LocalScale, discard return value
```

Comment:
This whole thing with localScale is a bit of a mess - not your fault of course. Could you possibly assign `localScale` in `BaseScreenComponent.SetParent` instead, or does that break things?

If not, at least add a  `UpdateLocalScale` function to call here rather than do the Property with side effects thing.

By the way, I'm curious, what's the code that relies on `localScale` in here? Not really obvious with all the Property assignments

Reply:
The LocalScale is used when calculating label length in TextLabel.CreateNewLabelLayoutWrapped().

I tried to make as low an impact change as I could, since we are post v1.0 now.  I'm reluctant to add code to SetParent due to the increased probability of inadvertently breaking something somewhere else.
I could add an UpdateLocalScale() method to TextLabel to make it clearer.